pkg/autoscaler/statforwarder/forwarder.go

```diff
@@ -311,16 +313,9 @@ func (f *Forwarder) createProcessor(bkt, holder string) *bucketProcessor {
 		}
 	}
 
-	// TODO(9235): Currently we use IP directly. This won't work if there
-	// is mesh. Need to fall back to connection via Service.
-	dns := fmt.Sprintf("ws://%s:%d", holder, autoscalerPort)
-	f.logger.Info("Connecting to Autoscaler bucket at ", dns)
-	return &bucketProcessor{
-		bkt:    bkt,
-		logger: f.logger.With(zap.String("bucket", bkt)),
-		holder: holder,
-		conn:   websocket.NewDurableSendingConnection(dns, f.logger),
-	}
+	return newForwardProcessor(f.logger.With(zap.String("bucket", bkt)), bkt, holder,
+		fmt.Sprintf("ws://%s:%d", holder, autoscalerPort),
+		fmt.Sprintf("ws://%s.%s.svc.%s:%d", bkt, ns, network.GetClusterDomainName(), autoscalerPort))
```

Comment:
```suggestion
		fmt.Sprintf("ws://%s.%s.svc.%s:%d", bkt, ns, network.GetClusterDomainName(), autoscalerPort))
```
`network.GetClusterDomainName(), autoscalerPort` pair won't change for the life of the process — probably we can format it once?